evap/staff/views.py

```diff
@@ -802,11 +814,62 @@ def questionnaire_copy(request, questionnaire_id):
             return render(request, "staff_questionnaire_form.html", dict(form=form, formset=formset))
     else:
         questionnaire = get_object_or_404(Questionnaire, id=questionnaire_id)
+        form, formset = get_identical_form_and_formset(questionnaire)
```

Comment:
while you are around here, could you move the get_object_or_404 to the beginning of the view? even if it's a POST it should crash if the url is invalid.